src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityRequest.java

```diff
@@ -202,6 +204,19 @@ private TenantSecurityException parseFailureFromRequest(HttpResponse resp) {
                 BatchUnwrappedDocumentKeys.class, error);
     }
 
+    /**
+     * Request re-key endpoint to unwrap an EDEK encrypted to the metadata's tenantId and then wrap it to the newTenantId.
+     */
+    CompletableFuture<RekeyedDocumentKey> rekey(String edek, DocumentMetadata metadata, String newTenantId) {
```

Comment:
If `edek` is a `String` going in, shouldn't we just give them a `String` going back out? I'm not sure I understand the value of the `RekeyedDocumentKey` vs using `String` or making a general `Edek` type that ensures `base64` and semantics across the many places we use edeks.

Reply:
Technically the TSP is sending back a full WrappedDocumentKey with DEK and EDEK, I just didn't think the DEK was necessary since it didn't change. Would it make more sense to return a `WrappedDocumentKey`? I wish I could make a type alias for it but I don't think I can

Reply:
I think this is fine. This isn't the public API so the "them" in this case is the `TenantSecurityClient`.